src/Components/Pages/Home.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState, useEffect } from 'react';
+import { AiOutlinePlus } from 'react-icons/ai';
+import Todo from '../Utilities/Todo';
+import Modal from '../Utilities/Modal';
+import '../Styles/Home.css';
+import { useModal } from '../Providers/ModalProvider';
+import { useTodos } from '../Providers/TodoProvider';
+
+const Home = () => {
+    const { modalOpen, setModalOpen } = useModal();
+    const { todos } = useTodos();
+    const [ rotation, setRotation ] = useState(0);
+    const [ searchQuery, setSearchQuery ] = useState('');
+
+    const toggleModal = e => {
+        setModalOpen(modalOpen.status === false ? {...modalOpen, status: true, data: { ...modalOpen.data, todoId: '', todoTitle: '', todoDesc: '', todoAlloted: '', todoTags: '', todoCreatedAt: ''  }} : {...modalOpen, status: false, data: { ...modalOpen.data, todoId: '', todoTitle: '', todoDesc: '', todoAlloted: '', todoTags: '', todoCreatedAt: ''  }})
```

Comment:
Instead of doing this. You can create a initial State which has these values. and one function to set These values. So if modal is open call that function to set that state. If its false, then instead of clearing all the fields manually you can directly assign initialState.